Update descriptions in `SecuritySettings`

systay opened the pull request:
`neo4j.conf` received some updates earlier, this PR updates the source code descriptions to match.

Commit:
Remove unnecessary output suppressions

Commit:
Update `@Description`s to conform to `neo4j.conf` docs

Minor wording fixes

fickludd commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -158,7 +160,7 @@
                   "an existing cache entry (instead of via an LDAP server) as long as it is alive " +
                   "(see `dbms.security.auth_cache_ttl`).\n" +
-                  "An important consequence of setting this to `true` than needs to be well understood, is that " +
+                  "An important consequence of setting this to `true` that needs to be well understood, is that " +
```
I think "that needs to be well understood" is redundant and could be removed all together

fickludd commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -170,7 +172,19 @@
     //-----------------------------------------------------
 
-    @Description( "Perform LDAP search for authorization info using a system account." )
+    @Description( "Perform LDAP search for authorization info using a system account or the user's own account.\n\n" +
```
maybe:
"Perform LDAP search for authorization info using a system account as opposed to the user's own account.\n\n"

craigtaverner replied:
Or: "Perform LDAP search for authorization info using a system account instead of the user's own account.\n\n"

craigtaverner replied:
Or:
"Perform LDAP search for authorization info using a system account instead of the user's own account.\n\n"

Mats-SX replied:
It took me a while to realise that @craigtaverner's suggestions were duplicate. I prefer that one here.

fickludd commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -170,7 +172,19 @@
+                  "group membership attributes on themselves in the directory.\n\n" +
+                  "If this is set to `true`, the search will be performed using a special system account user " +
+                  "with read access to all the users in the directory. " +
+                  "You need to specify the username and password with the next two settings below with his option. " +
```
instead of saying "the next two settings", these could be named

jjaderberg replied:
If "the next two settings" means making assumptions on order in the resulting content then indeed the two settings need to be referenced. Other components may have settings that are inserted in between, or the principle of order and grouping may change.

Mats-SX replied:
And both of you missed that it says `his` rather than `this` :D

Mats-SX replied:
I changed it to the following:

> You need to specify the username and password using the settings `dbms.security.ldap.authorization.system_username` and `dbms.security.ldap.authorization.system_password` with this option.

fickludd commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -234,7 +254,10 @@
+    @Description( "Set to log successful authentication events to the security log." +
+                  "If this is set to `false` only failed authentication events will be logged, which " +
+                  "could be useful if you find that the successful events spam the logs too much, " +
+                  "and you do not require full auditing capability")
```
codestyle?

henriknyman replied:
Also missing a space after the . on the first line of the description

jjaderberg replied:
Code style _ought_ to be: "One sentence per line." when writing docs, even if the docs happen to be embedded in other code. Not holding my breath about that one, though.

Mats-SX replied:
@jjaderberg Judging by the looks of the `@Description`s in `GraphDatabaseSettings`, that is _not_ current styling. It seems these are optimised for code readability, and to fit on 120 chars.

If this ought to change, I think we should have a conscious effort of adapting all settings to that standard.

henriknyman commented on packaging/standalone/standalone-enterprise/src/main/distribution/text/enterprise/conf/neo4j.conf:
```diff
@@ -430,14 +430,15 @@ dbms.connector.https.enabled=true
 # E.g. group1=role1;group2=role2;group3=role3,role4,role5
-#dbms.security.ldap.authorization.group_to_role_mapping=
-
+#
```
I think according to the policy this should be here to show that the default value is empty

fickludd commented on packaging/standalone/standalone-enterprise/src/main/distribution/text/enterprise/conf/neo4j.conf:
```diff
@@ -373,7 +373,7 @@ dbms.connector.https.enabled=true
 # an existing cache entry (instead of via an LDAP server) as long as it is alive
 # (see `dbms.security.auth_cache_ttl`).
-# An important consequence of setting this to `true` than needs to be well understood, is that
+# An important consequence of setting this to `true` that needs to be well understood, is that
```
Same "that needs to be well understood" comment as above

fickludd commented:
left some minor nitpicks, take or leave

henriknyman commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -189,7 +203,7 @@
 
     @Description( "The name of the base object or named context to search for user objects when LDAP authorization is " +
-                  "enabled." )
+                  "enabled. A common case is that this matches the last part of `dbms.security.ldap.user_dn_template`." )
```
This setting is now called `dbms.security.ldap.authentication.user_dn_template`

henriknyman replied:
This is also wrong in two places in the comments in neo4j.conf. Maybe you could fix it in this PR?

henriknyman replied:
Also it is incorrect in the description for `dbms.security.ldap.authorization.system_username` above this

craigtaverner commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -115,12 +115,14 @@
     @Description( "URL of LDAP server (with protocol, hostname and port) to use for authentication and authorization. " +
                   "If no protocol is specified the default will be `ldap://`. To use LDAPS, " +
-                  "set the protocol and port, e.g. `ldaps://ldap.example.com:636`" )
+                  "set the protocol and port, e.g. `ldaps://ldap.example.com:636` " +
```
You do not need to set the port for LDAPS, it defaults to 636.

jjaderberg replied:
If the default is 636 why is the default setting value `localhost:389`?

jjaderberg replied:
Even if 636 is the default value it is fine (safe) to use it in an example.

Mats-SX replied:
I guess it's default `389` because the default is to use `LDAP` non-secure, rather than `LDAPS`? Please confirm @craigtaverner @henriknyman 

@craigtaverner Does changing the text to the following satisfy your comment?

> To use LDAPS, set the protocol to `ldaps://ldap.example.com`. The default port for LDAPS is 636.

craigtaverner commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -115,12 +115,14 @@
+                  "NOTE: You may want to consider using STARTTLS (`dbms.security.ldap.use_starttls`) over LDAPS " +
+                  "for secure connections, in which case the default port will be sufficient.")
     public static final Setting<String> ldap_server =
             setting( "dbms.security.ldap.host", STRING, "localhost:389" );
```
You do not need to set the port, it defaults to 389 for LDAP and 636 for LDAPS.

Mats-SX replied:
Are you suggesting we change the default value to not mention the port because there is code that falls back to 389?

jjaderberg commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -129,7 +131,7 @@
             "The LDAP referral behavior when creating a connection. This is one of `follow`, `ignore` or `throw`.\n" +
```
Bulleted lists don't work in this context. The output will be a paragraph with some asterisks in it. It's out of scope of this PR to fix this, but worth noting that adding `* item` won't actually add items to a proper list.

Mats-SX replied:
Would you advise to move it into a normal sentence without line breaks and asterisks then? It's easy to fix.

jjaderberg commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -170,7 +172,19 @@
+                  "and then expire, requiring re-authentication. To avoid frequently having to re-authenticate " +
+                  "sessions you may want to set a relatively long auth cache expiration time together with this " +
+                  "option. NOTE: This option will only work if the users are permitted to search for the " +
+                  "group membership attributes on themselves in the directory.\n\n" +
```
"for the group membership attributes on themselves" => "for their own group memberships" or "for group memberships for their own user"

Mats-SX replied:
I changed it to

> for their own group membership attributes in the directory

jjaderberg commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -170,7 +172,19 @@
+                  "If this is set to `true`, the search will be performed using a special system account user " +
+                  "with read access to all the users in the directory. " +
+                  "You need to specify the username and password with the next two settings below with his option. " +
+                  "Note that this account only needs read-only access to the relevant parts of the LDAP directory " +
```
Can we truthfully get away with "read access" to avoid the double "only"?

Mats-SX replied:
I think so, yes.

jjaderberg commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -170,7 +172,19 @@
+                  "with read access to all the users in the directory. " +
+                  "You need to specify the username and password with the next two settings below with his option. " +
+                  "Note that this account only needs read-only access to the relevant parts of the LDAP directory " +
+                  "and does not need to have access rights to Neo4j or any other systems." )
```
"Neo4j, or any other"

jjaderberg commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -204,19 +218,25 @@
+                  "The map should be formatted as a semicolon separated list of key-value pairs, where the " +
                   "key is the LDAP group name and the value is a comma separated list of corresponding role names. " +
-                  "E.g. group1=role1;group2=role2;group3=role3,role4,role5" )
+                  "E.g. group1=role1;group2=role2;group3=role3,role4,role5\n\n" +
```
Prefer "For example," over "E.g.,"

jjaderberg commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -204,19 +218,25 @@
-    @Description( "The time to live (TTL) for cached authentication and authorization info. Setting the TTL to 0 will" +
-            " disable auth caching." )
+    @Description( "The time to live (TTL) for cached authentication and authorization info when using " +
+                  "external auth providers (ldap or plugin). Setting the TTL to 0 will disable auth caching." )
```
capitalize LDAP

Mats-SX commented:
Thanks for all the great feedback guys!

Mats-SX committed:
Fixes from review

- `e.g.` -> `for example`
- Fix mention of configs that have changed names
- capitalize `LDAP`
- language improvements

Mats-SX commented:
The GitHub UI hides it, but there are two additional outstanding comments by @craigtaverner that I haven't yet addressed.

Mats-SX committed:
Clarify usage of `dbms.security.ldap.host`

henriknyman commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -234,7 +259,10 @@
+    @Description( "Set to log successful authentication events to the security log. " +
+                  "If this is set to `false` only failed authentication events will be logged, which " +
+                  "could be useful if you find that the successful events spam the logs too much, " +
+                  "and you do not require full auditing capability" )
```
Missing . at the end of sentence.

henriknyman commented on packaging/standalone/standalone-enterprise/src/main/distribution/text/enterprise/conf/neo4j.conf:
```diff
@@ -334,11 +334,14 @@ dbms.connector.https.enabled=true
-# NOTE: You may want to consider using STARTTLS (`dbms.security.ldap.use_starttls`) over LDAPS
-# for secure connections, in which case the default port will be sufficient.
-#dbms.security.ldap.host=localhost:389
+# The format of the setting is `<protocol>://<hostname>:<port>`, where hostname is the only required field.
```
This is a bit chatty now, maybe "(with protocol, hostname and port)" can go away since it is made redundant by this line.

henriknyman commented on packaging/standalone/standalone-enterprise/src/main/distribution/text/enterprise/conf/neo4j.conf:
```diff
@@ -334,11 +334,14 @@ dbms.connector.https.enabled=true
+# The format of the setting is `<protocol>://<hostname>:<port>`, where hostname is the only required field.
+# The supported values for protocol are `ldap` (default) and `ldaps`.
+# The default port for `ldap` is 389 and for `ldaps` 636.
+# For example: `ldaps://ldap.example.com:4444`.
```
4444 is a bit strange example, or is it some kind of standard example? Otherwise 10389 could be more of a real world use case example

Mats-SX replied:
No we just picked a number at random. Changed it to 10389.

henriknyman commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -204,19 +223,25 @@
+                  "         \"cn=Neo4j Read Only,cn=users,dc=example,dc=com\"      = reader;    \\\n" +
+                  "         \"cn=Neo4j Read-Write,cn=users,dc=example,dc=com\"     = publisher; \\\n" +
+                  "         \"cn=Neo4j Schema Manager,cn=users,dc=example,dc=com\" = architect; \\\n" +
+                  "         \"cn=Neo4j Administrator,cn=users,dc=example,dc=com\"  = admin" )
```
Is this formatting presented correctly in the manual? Otherwise maybe it is enough to have it in `neo4j.conf` only.

Mats-SX replied:
I think it respects the newlines. I'd rather not have to redo this one again.

henriknyman commented on enterprise/security/src/main/java/org/neo4j/server/security/enterprise/configuration/SecuritySettings.java:
```diff
@@ -114,13 +114,17 @@
     @Description( "URL of LDAP server (with protocol, hostname and port) to use for authentication and authorization. " +
-                  "If no protocol is specified the default will be `ldap://`. To use LDAPS, " +
-                  "set the protocol and port, e.g. `ldaps://ldap.example.com:636`" )
+                  "The format of the setting is `<protocol>://<hostname>:<port>`, where hostname is the only " +
```
This is a bit chatty now, maybe "(with protocol, hostname and port)" can go away since it is made redundant by this line.

Mats-SX committed:
Change example port

Correct language error.

Mats-SX commented:
@jjaderberg Is this PR approved, or are there more changes that you'd like to see?
